Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/AppHost/ConsoleWindow.tsx

```diff
@@ -92,6 +95,34 @@ const ConsoleWindow = (props: IProps) => {
     }, [update]);
 
 
+    React.useEffect(() => {
+        if (props.ConsoleURL != undefined && props.ConsoleURL.length > 0) 
+            document.addEventListener("keydown", handleKeyPress, false);
+
+        return () => {
+            document.removeEventListener("keydown", handleKeyPress, false);
+        }
+    }, [props.ConsoleURL])
+
+    React.useEffect(() => { cmdRef.current = cmd; }, [cmd]);
+    React.useEffect(() => { lastCmdRef.current = lastCmd; }, [lastCmd]);
+    
+    function handleKeyPress(event) {
+
+        if (event.keyCode == 38 && cmd.length > 0) // arrow down key
+        {
+            event.preventDefault();
+            setCMD(cmdRef.current); 
```

Comment:
It looks like we are setting cmd to cmdRef.current, which the use effect sets to cmd. I don't see anywhere else that sets this ref. Am I missing something?